tests/test_modeling_common.py

```diff
@@ -693,6 +694,107 @@ def test_forward_signature(self):
+            elif isinstance(batched_object, Dict):
+                for batched_object_value, single_row_object_value in zip(
+                    batched_object.values(), single_row_object.values()
+                ):
+                    recursive_check(batched_object_value, single_row_object_value, model_name, key)
+            elif batched_object is None or isinstance(batched_object, int):
+                return
+            elif batched_object.dim() == 0:
+                return
+            else:
+                # indexing the first element does not always work
+                # e.g. models that output similarity scores of size (N, M) would need to index [0, 0]
+                slice_ids = [slice(0, index) for index in single_row_object.shape]
+                batched_row = batched_object[slice_ids]
+                self.assertFalse(
+                    torch.isnan(batched_row).any(), f"Batched output has `nan` in {model_name} for key={key}"
+                )
+                self.assertFalse(
+                    torch.isinf(batched_row).any(), f"Batched output has `inf` in {model_name} for key={key}"
+                )
+                self.assertFalse(
+                    torch.isnan(single_row_object).any(), f"Single row output has `nan` in {model_name} for key={key}"
+                )
+                self.assertFalse(
+                    torch.isinf(single_row_object).any(), f"Single row output has `inf` in {model_name} for key={key}"
+                )
+                self.assertTrue(
+                    equivalence(batched_row, single_row_object) <= 1e05,
+                    msg=(
+                        f"Batched and Single row outputs are not equal in {model_name} for key={key}. "
+                        f"Difference={equivalence(batched_row, single_row_object)}."
+                    ),
+                )
+
+        config, batched_input = self.model_tester.prepare_config_and_inputs_for_common()
+        equivalence = get_tensor_equivalence_function(batched_input)
+
+        for model_class in self.all_model_classes:
+            config.output_hidden_states = True
+            config.output_attentions = True
+
+            model_name = model_class.__name__
+            if hasattr(self.model_tester, "prepare_config_and_inputs_for_model_class"):
+                config, batched_input = self.model_tester.prepare_config_and_inputs_for_model_class(model_class)
+            batched_input_prepared = self._prepare_for_class(batched_input, model_class)
+            model = model_class(config).to(torch_device).eval()
+
+            # We can't infer batch size from input shape, inputs can have multiple vectors for each batch,
+            # so we try to get it from model_tester first. Some composite models, e.g. Owlv2ModelTest,
+            # do not indicate batch size in model testers, but rather in the parent's tester, then we
+            # try to get parent tester's batch size
```

Comment:
It seems like an easier fix would be to unify the model testers to have the batch size as standard

Reply:
hmm, okay, I will try

Reply:
Done and checked for all models and testers, not only the current test. Also, I added an error raising if the batch size is not 14 for any future model tests

Reply:
That is the largest size I found from existing tests. If it is too big for tests, I can try something like 8 and test again, that nothing breaks. The only constraint I saw was that some models need an even batch size

Reply:
Two questions:
- Why are we inforcing here for all the model testers to have the same value?
- Why take the max instead of the min?

Reply:
- If a new model test is added and it does not use the standard batch size, the model might fail on the `batching_equivalence` test due to incorrectly inferred batch size (which is now not inferred from tester, but fixed to 14). As an option, we can fall back to taking the first row as one-batch input, so raising error will not not needed. But for particular models, where each batch can consist of multiple rows, the tests will fail anyway if they use anything expect for the standard 14. I decided to force the standard size, so that we do not have to worry about various batch size causing this test to fail in the future.
I am open to remove enforcing, and the future tests in case of failing will adjust batch sizes. Anyway, now I realize that most models have a standard one-bacth=one-row format :) 

- The max was chose for no particular reason, since I needed to have some guidance on which unified batch size to use for the test. Also, I was not sure if some model's tests rely heavily on larger batch size, so the max seemed like the better choice than min or avg

Reply:
Sorry, I still don't understand why is has to be a fixed batch size for all models for this test. 

> But for particular models, where each batch can consist of multiple rows, the tests will fail anyway if they use anything expect for the standard 14.

Why is this the case i.e. what is it about the multiple rows that's causing it to fail? Is it specifically a batch size of 14 where this is true? 

The only thing I think we should be enforcing is that batch_size > 1. 

For some models, if they have special circumstances, then we can set `batch_size=14` for them. If there's additional  special logic required for those models, then they should have their own implementation `test_batching_equivalence` added in their model's test file

Reply:
> It seems like an easier fix would be to unify the model testers to have the batch size as standard 

Maybe I misunderstood your first suggestion. I thought you suggested to use a fixed batch size for all testers, i.e. using a fixed batch size for this test, consequently for all testers.

> Why is this the case i.e. what is it about the multiple rows that's causing it to fail? Is it specifically a batch size of 14 where this is true?

Regarding this, it does not fail now, neither it failed before the last commit. It has nothing to do which batch size we use, but rather if we can correctly infer the batch size from data/testers. 

It will not fail for any future test also, but only if we infer the correct batch size, because we need to dissect out one batch out of all fake-data. If we do not succeed to dissect exactly one batch, the hidden states will result having high differences, the test fails.

Before "standard" batch size of 14, this case was handled by trying to infer `bs` from testers, but now we don't infer. Therefore, I decided to enforce, so no-one spends time to find the reason of same error again in the future. As another option, I can add a comment, not error about this case.

I hope it is clearer, but if not we can talk through it in a call. I guess that would be faster :D

Reply:
> Maybe I misunderstood your first suggestion. I thought you suggested to use a fixed batch size for all testers, i.e. using a fixed batch size for this test, consequently for all testers.

@zucchini-nlp Ah, OK, sorry, I wasn't clear. What I meant was, we can just make sure all model testers have `batch_size` available for all model testers rather than having to search at different levels for composite models

Reply:
Oh, that makes more sense. Got it, I will revert the changes back

Reply:
done, added bs for composite testers where missing :)